### PR TITLE
register_hash

### PR DESCRIPTION
**Pull request by kuanyiwu**
Added register options to jsps.
Added BCrypt for password hashing.
Commented out some tests in LoginServletTest.

### CODE REVIEW

File: src/main/java/codeu/controller/LoginServlet.java [hunk 1]
```diff
@@ -14,16 +14,18 @@
...
+
 import javax.servlet.ServletException;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+import org.mindrot.jbcrypt.BCrypt;
+
+import codeu.model.data.User;
+import codeu.model.store.basic.UserStore;
```
Comment (aditija) [inline src/main/java/codeu/controller/LoginServlet.java]: In general it's good to keep imports alphabetical

File: src/main/java/codeu/controller/LoginServlet.java [hunk 2]
```diff
@@ -63,45 +65,29 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
...
-  }
+ @Override
+ public void doPost(HttpServletRequest request, HttpServletResponse response)
+     throws IOException, ServletException {
+   String username = request.getParameter("username");
+   String password = request.getParameter("password");
 
+   if (userStore.isUserRegistered(username)) {
+     User user = userStore.getUser(username);
+     if(BCrypt.checkpw(password, user.getPassword())) {
```
Comment (aditija) [inline src/main/java/codeu/controller/LoginServlet.java]: Add space between 'if' and '('

File: src/test/java/codeu/controller/LoginServletTest.java [hunk 3]
```diff
@@ -103,6 +106,6 @@ public void testDoPost_ExistingUser() throws IOException, ServletException {
     Mockito.verify(mockSession).setAttribute("user", "test username");
     Mockito.verify(mockResponse).sendRedirect("/conversations");
     //check if password is correct
-//    Mockito.verify("password", Mockito.getPassword());
-  }
+    //Mockito.verify("password", Mockito.getPassword());
```
Comment (aditija) [inline src/test/java/codeu/controller/LoginServletTest.java]: Why do these tests need to be commented out?

Comment (kuanyiwu) [reply src/test/java/codeu/controller/LoginServletTest.java]: These tests do not pass after the register servlet is added.
I can try to see what needs to be changed.

Comment (kuanyiwu) [reply src/test/java/codeu/controller/LoginServletTest.java]: Project 1-7 modify loginservlet
This change will now cause the LoginServletTest to fail. Fix that test before continuing. Remove tests that are no longer needed, and add new tests that check the different kinds of input the LoginServlet can receive.

Comment (kuanyiwu) [reply src/test/java/codeu/controller/LoginServletTest.java]: Okay so I read through the tests and before the login servlet is also performing functions of register servlets, and that's what these tests are for. Now register test handles that so these tests are not needed anymore.

**Pull request merged**